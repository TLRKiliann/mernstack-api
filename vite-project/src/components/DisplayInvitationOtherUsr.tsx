import React from 'react'

interface DisplayInvitationOtherUsrProps {
  isCheckInvite: boolean
  handleSwitchBox: () => void
  handleInvitedResponse: (e: React.MouseEvent<HTMLButtonElement>) => void
}

const DisplayInvitationOtherUsr: React.FC = (props: DisplayInvitationOtherUsrProps) => {
  return (
    <div>
      <h4 className="invitation--display">You've recieved invitation from {props.username}.</h4>
      <h4 className="invitation--display">Are you ok for {} chat ?</h4>
      <label>
        <input
          type="checkbox"
          checked={props.isCheckInvite}
          onChange={props.handleSwitchBox}
        />
        yes
      </label>
      <button
        type="button"
        onClick={props.handleInvitedResponse}
      >
        Validate
      </button>
    </div>
  )
}

export default DisplayInvitationOtherUsr