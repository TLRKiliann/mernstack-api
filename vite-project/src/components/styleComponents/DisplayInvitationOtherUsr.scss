.invitation--display {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 400px;
  height: 400px;
  margin: auto;
  margin-top: 20%;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  background: linear-gradient(180deg,
    orange, purple);
  border-radius: 15px;
  z-index: 2;
}

.invitation--h4 {
  color: whitesmoke;
}

.btn--displayotheruser {
  margin-top: 20px;
  margin-bottom: 20px;
  padding: 5px 10px;
  font-family: "DejaVu Sans", sans-serif;
  font-size: 0.9rem;
  font-weight: bold;
  background: linear-gradient(180deg,
    orange, orangered, #ae3c52);
  border: 1px solid purple;
  border-radius: 7px;
  color: whitesmoke;
  &:hover {
    background: orange;
    color: #ae3c52;
  }
}