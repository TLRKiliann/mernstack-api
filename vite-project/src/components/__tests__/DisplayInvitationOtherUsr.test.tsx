import React from "react";
import { screen, render } from '@testing-library/react';
import { test, expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import DisplayInvitationOtherUsr from "../DisplayInvitationOtherUsr.tsx";

test('DisplayInvitationOtherUsrmy is present', () => {
  render(<DisplayInvitationOtherUsr />);
  const displayOthertestid = screen.getByTestId("displayOthertestid");
  expect(displayOthertestid).toBeInTheDocument();
})

test('MatchSnapShot test DisplayInvitationOtherUsr', () => {
  const treeDisplayInvitationOtherUsr = create(<DisplayInvitationOtherUsr />)
  expect(treeDisplayInvitationOtherUsr.toJSON()).toMatchSnapshot()
})