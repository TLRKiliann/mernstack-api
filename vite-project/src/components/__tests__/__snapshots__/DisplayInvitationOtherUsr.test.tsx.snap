// Vitest Snapshot v1

exports[`MatchSnapShot test DisplayInvitationOtherUsr 1`] = `
<div
  className="invitation--display"
  data-testid="displayOthertestid"
>
  <h4
    className="invitation--h4"
  >
    <span
      className="span--h4"
    >
       !
    </span>
    <p
      style={
        {
          "marginTop": "30px",
        }
      }
    >
      You've recieved invitation from
    </p>
    <span
      className="span--h4"
    >
      .
    </span>
  </h4>
  <div
    className="definedroom--display"
  >
    <p
      className="invitation--p"
    >
      Are you ok for
      <span
        className="span--invitationp"
      />
      chat ?
    </p>
    <div
      className="checkbox--firstconfirm"
    >
      <label
        className="checkbox--lbl"
      >
        <input
          type="checkbox"
        />
         Yes
      </label>
      <label
        className="checkbox--lbl"
      >
        <input
          type="checkbox"
        />
         No
      </label>
    </div>
  </div>
  <button
    className="btn--displayotheruser"
    type="button"
  >
    Validate
  </button>
</div>
`;
